conf/modules.config

```
@@ -12,28 +12,42 @@
 
 process {
 
-    withName: FASTP {
+    withName: "AMPLICON_PIPELINE:READS_QC_MERGE:FASTP" {
         publishDir = [
             [
                 path: { "${params.outdir}/${meta.id}/qc/" },
                 mode: params.publish_dir_mode,
                 failOnError: true,
-                pattern: "*[!_{1,2}].fastp.fastq.gz"
+                saveAs: {
+                    filename ->
+                    {
+                        if ( meta.single_end ) {
+                            if ( filename.contains(".fastp.fastq.gz") ){
+                                return filename;
+                            }
+                            else{
+                                return null;
+                            }
+                        }
+                        else {
+                            if ( filename.contains(".merged.fastq.gz") ){
+                                return filename;
+                            }
+                            else{
+                                return null;
+                            }
+                        }
+                    }
+                }
             ],
             [
                 path: { "${params.outdir}/${meta.id}/qc/" },
                 mode: params.publish_dir_mode,
                 failOnError: true,
-                pattern: "*.merged.fastq.gz"
-            ],
-            [
-                path: { "${params.outdir}/${meta.id}/qc/" },
-                mode: params.publish_dir_mode,
-                failOnError: true,
-                pattern: ".json" 
+                pattern: "*.json"
             ]
         ]
-    ext.args = '-l 100' // Filter out reads smaller than 100 bases
+    ext.args = '--cut_front_window_size 1 --cut_tail_window_size 1 --cut_front 3 --cut_tail 3 --cut_right --cut_right_window_size 4 --cut_right_mean_quality 15 -l 100' // Filter out reads smaller than 100 bases
```

```suggestion
    ext.args = [
    '--cut_front_window_size 1',
    '--cut_tail_window_size 1',
    '--cut_front 3',
    '--cut_tail 3',
    '--cut_right',
    '--cut_right_window_size 4',
    '--cut_right_mean_quality 15',
    '-l 100' // Filter out reads smaller than 100 bases
    ].join(' ');
```
Again, this is just to make it easier to spot which options we are using. One flag with its value per line makes it easier to identify.